tools/base/fpgainfo/board.c

```diff
@@ -0,0 +1,448 @@
+
+// Board plug-in table
+static platform_data platform_data_table[] = {
+	{ 0x8086, 0x09c4, "libboard_rc.so", NULL },
+	{ 0x8086, 0x09c5, "libboard_rc.so", NULL },
+	{ 0,      0,          NULL, NULL },
+};
+
+
+fpga_result load_board_plugin(fpga_token token, void** dl_handle)
+{
+	fpga_result res                = FPGA_OK;
+	fpga_result resval             = FPGA_OK;
+	fpga_properties props          = NULL;
+	uint16_t vendor_id             = 0;
+	uint16_t device_id             = 0;
+	int i                          = 0;
+
+	if (token == NULL || dl_handle == NULL) {
+		OPAE_ERR("Invalid input parameter");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaGetProperties(token, &props);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get properties\n");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaPropertiesGetDeviceID(props, &device_id);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get device ID\n");
+		resval = res;
+		goto destroy;
+	}
+
+	res = fpgaPropertiesGetVendorID(props, &vendor_id);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get vendor ID\n");
+		resval = res;
+		goto destroy;
+	}
+
+	if (pthread_mutex_lock(&board_plugin_lock) != 0 ) {
+		OPAE_ERR("pthread mutex lock failed \n");
+		resval = FPGA_EXCEPTION;
+		goto destroy;
+	}
+
+	for (i = 0; platform_data_table[i].board_plugin; ++i) {
+
+		if (platform_data_table[i].device_id == device_id &&
+			platform_data_table[i].vendor_id == vendor_id) {
+
+			// Loaded lib or found
+			if (platform_data_table[i].dl_handle) {
+				*dl_handle = platform_data_table[i].dl_handle;
+				resval = FPGA_OK;
+				goto unlock_destroy;
+			}
+
+			platform_data_table[i].dl_handle = dlopen(platform_data_table[i].board_plugin, RTLD_LAZY | RTLD_LOCAL);
+			if (!platform_data_table[i].dl_handle) {
+				char *err = dlerror();
+				OPAE_ERR("Failed to load \"%s\" %s", platform_data_table[i].board_plugin, err ? err : "");
+				resval = FPGA_EXCEPTION;
+				goto unlock_destroy;
+			} else {
+				// Dynamically loaded board module
+				*dl_handle = platform_data_table[i].dl_handle;
+				resval = FPGA_OK;
+				goto unlock_destroy;
+			}
+		} //end if
+
+	} // end for
+
+
+unlock_destroy:
+
+	if (pthread_mutex_unlock(&board_plugin_lock) != 0) {
+		OPAE_ERR("pthread mutex unlock failed \n");
+		resval = FPGA_EXCEPTION;
+	}
+
+destroy:
+	res = fpgaDestroyProperties(&props);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Destroy Object");
+	}
+
+	return resval;
+}
+
+int unload_board_plugin(void)
+{
+	int i            = 0;
+	int res          = 0;
+
+	if (pthread_mutex_lock(&board_plugin_lock) != 0) {
+		OPAE_ERR("pthread mutex lock failed \n");
+		return FPGA_EXCEPTION;
+	}
+
+	for (i = 0; platform_data_table[i].board_plugin; ++i) {
+
+		if (platform_data_table[i].dl_handle) {
+
+			res = dlclose(platform_data_table[i].dl_handle);
+			if (res) {
+				char *err = dlerror();
+				OPAE_ERR("dlclose failed with %d %s", res, err ? err : "");
+				res = FPGA_EXCEPTION;

```

Comment:
This will be overwritten by the next iteration.

Reply:
fixed